### DISCUSSION TITLE
How can I add a custom mime type for rendering a separate output

### DISCUSSION THREAD

**Discussion by savakarrohan** [jupyterlite/jupyterlite#1224] [General]
Hey, I have some python calculations which are being done and all the data is being stored as possibly a `json` or a `string` object. 
I would like to create a custom renderer for this, such that it reads this json object and renders a `react widget`.
How can I do the same?
For example I have: 
![image](https://github.com/jupyterlite/jupyterlite/assets/50200523/182ebec4-9bd3-410f-a677-552326abaf4a)
The above `json` saved in a variable, and I would like to render it using my own react widget rather than the `text/plain` method shown below using `kernelSpy`
![image](https://github.com/jupyterlite/jupyterlite/assets/50200523/b3c384ff-45fb-4779-be48-127ea1f66f9b)

**Comment by jtpio** [comment]
Maybe the best would be create a mimerender extension from the JupyterLab template? https://github.com/jupyterlab/extension-template

And include that new mimerender extension in a JupyterLite deployment.

**Reply by savakarrohan** [reply]
Thank you!!!
I managed to do it, and it works fabulously now!
